### PR TITLE
HEAT-282  Adding lists of components missing teams

### PR DESCRIPTION
**Pull request by andrewrlee**
incl:
* Adding components missing teams
* Fixing security vuln
* Adding component list to service area view

### CODE REVIEW

File: server/utils/nunjucksSetup.ts [hunk 1]
```diff
@@ -53,6 +53,11 @@ export default function nunjucksSetup(app: express.Express, applicationInfo: App
 
   njkEnv.addFilter('toMonitorName', (val: string) => formatMonitorName(val))
 
+  njkEnv.addFilter(
```
Comment (andrewrlee) [inline server/utils/nunjucksSetup.ts:56]: little util to help developing templates, this filter pretty prints the model in nested json

File: server/views/pages/product.njk [hunk 2]
```diff
@@ -23,7 +23,7 @@
     </tr>
     <tr>
       <th>ID</th>
-      <td>{{ product.id }}</p></td>
+      <td>{{ product.id }}</td>
```
Comment (andrewrlee) [inline server/views/pages/product.njk:26]: This led to weirdly formatted data:
<kbd>
Screenshot 2024-08-29 at 12 15 51
</kbd>

File: server/views/pages/serviceArea.njk [hunk 3]
```diff
@@ -25,7 +25,14 @@
...
         <ul>
           {% for product in serviceArea.products %}
-            <li><a href="/products/{{ product.attributes.slug }}" data-test="product-{{ product.id }}">{{ product.attributes.name }}</a></li>
+            <li>
+              <a class="govuk-link--no-visited-state govuk-!-margin-bottom-0 govuk-!-margin-top-3 govuk-heading-s" href="/products/{{ product.attributes.slug }}" data-test="product-{{ product.id }}">{{ product.attributes.name }}</a>
+              <ul>
+              {% for component in product.attributes.components.data %}
+                <li><a class="govuk-link--no-visited-state" href="/components/{{component.attributes.name}}" data-test="product-{{ product.id }}-component-{{ component.id }}">{{component.attributes.name}}</a></li>
+              {% endfor %} 
+              </ul>
```
Comment (andrewrlee) [inline server/views/pages/serviceArea.njk:34]: Adding components to service area
<kbd>
Screenshot 2024-08-29 at 12 18 10

</kbd>

File: server/views/pages/missingFromCatalogue.njk [hunk 4]
```diff
@@ -53,6 +52,35 @@
...
+            {
+            text: "Component"
+            },
+            {
+            text: "Product Missing Team"
+            }
+        ],
+        rows: rows
+    }) }}
+ {% endset %}
```
Comment (andrewrlee) [inline server/views/pages/missingFromCatalogue.njk:81]: Adding new view:

<kbd>

Screenshot 2024-08-29 at 12 19 37

</kbd>

**Review by mattops — APPROVED** [review]

**Pull request merged**